File: JAVA/Aksesoris.java
// Kelas Aksesoris merupakan turunan dari PetShop
public class Aksesoris extends PetShop {
    // Deklarasi atribut tambahan untuk aksesoris
    private String jenis;
    private String bahan;
    private String warna;

    // Constructor default untuk inisialisasi atribut dengan nilai awal
    public Aksesoris() {
        super();            // Memanggil constructor default dari superclass (PetShop)
        this.jenis = "";
        this.bahan = "";
        this.warna = "";
    }

    // Constructor dengan parameter untuk mengisi atribut saat objek dibuat
    public Aksesoris(String id, String namaProduk, int hargaProduk, int stokProduk, String jenis, String bahan, String warna) {
        super(id, namaProduk, hargaProduk, stokProduk); // Memanggil constructor superclass dengan parameter
        this.jenis = jenis;
        this.bahan = bahan;
        this.warna = warna;
    }

    // Getter untuk mengambil nilai atribut jenis
    public String getJenis() {
        return jenis;
    }

    // Setter untuk mengubah nilai atribut jenis
    public void setJenis(String jenis) {
        this.jenis = jenis;
    }

    // Getter untuk mengambil nilai atribut bahan
    public String getBahan() {
        return bahan;
    }

    // Setter untuk mengubah nilai atribut bahan
    public void setBahan(String bahan) {
        this.bahan = bahan;
    }

    // Getter untuk mengambil nilai atribut warna
    public String getWarna() {
        return warna;
    }

    // Setter untuk mengubah nilai atribut warna
    public void setWarna(String warna) {
        this.warna = warna;
    }

    // Overriding method display untuk menampilkan informasi produk aksesoris
    public void display(int idMax, int namaMax, int hargaMax, int stokMax, int jenisMax, int bahanMax, int warnaMax) {
        super.display(idMax, namaMax, hargaMax, stokMax); // Menampilkan data dari superclass
        System.out.printf("%-" + jenisMax + "s %-" + bahanMax + "s %-" + warnaMax + "s\n", jenis, bahan, warna); // Menampilkan atribut tambahan
    }
}

File: PYTHON/Aksesoris.py
from PetShop import PetShop  # Mengimpor kelas induk PetShop

class Aksesoris(PetShop):  # Membuat subclass Aksesoris yang mewarisi PetShop
    
    def __init__(self, id="", nama_produk="", harga_produk=0, stok_produk=0, jenis="", bahan="", warna=""):
        super().__init__(id, nama_produk, harga_produk, stok_produk)  # Memanggil konstruktor kelas induk
        # Konstruktor untuk menginisialisasi atribut dengan nilai default
        self.set_jenis(jenis)  
        self.set_bahan(bahan)  
        self.set_warna(warna)  

    def set_jenis(self, jenis):
        self._jenis = jenis  # Menyimpan jenis aksesoris
    
    def get_jenis(self):
        return self._jenis  # Mengembalikan jenis aksesoris

    def set_bahan(self, bahan):
        self._bahan = bahan  # Menyimpan bahan aksesoris
    
    def get_bahan(self):
        return self._bahan  # Mengembalikan bahan aksesoris
    
    def set_warna(self, warna):
        self._warna = warna  # Menyimpan warna aksesoris
    
    def get_warna(self):
        return self._warna  # Mengembalikan warna aksesoris
    
    def display(self, jenis_max, bahan_max, warna_max):
        # Menampilkan data produk dengan format yang rapi berdasarkan panjang maksimal kolom
        print(
            self._jenis + " " * (jenis_max - len(self._jenis)),
            self._bahan + " " * (bahan_max - len(self._bahan)),
            self._warna + " " * (warna_max - len(self._warna))
        )
